### PR TITLE
Extend control elements with custom classNames

### PR DESCRIPTION
**Pull request by rshtg**
Re-opened [#376](https://github.com/uber/react-map-gl/pull/376)

Added extended prop `className` for:
* `Marker`
* `Popup`
* `NavigationControl`

### CODE REVIEW

File: src/components/marker.js [hunk 1]
```diff
@@ -53,6 +53,8 @@ export default class Marker extends Component {
   render() {
     const {longitude, latitude, offsetLeft, offsetTop} = this.props;
 
+    let className = this.props.className || '';
```
Comment (Pessimistress) [inline src/components/marker.js]: You can just add `className: ''` to `defaultProps`.

**Comment by Pessimistress** [comment]
@rshtg Could you resolve the conflicts? We can publish a patch once this is merged.

**Comment by rshtg** [comment]
@Pessimistress It seems like CI didn't completed. Unfortunately I can't to understand why it causes. Could you clarify if possible please?

File: src/components/navigation-control.js [hunk 2]
```diff
@@ -77,8 +78,11 @@ export default class NavigationControl extends BaseControl {
   }
 
   render() {
+
+    const { className } = this.props;
```
Comment (ibgreen) [inline src/components/navigation-control.js]: This is your CI error
```
/home/travis/build/uber/react-map-gl/src/components/navigation-control.js
  82:11  error  There should be no space after '{'   object-curly-spacing
  82:23  error  There should be no space before '}'  object-curly-spacing
```

**Pull request merged**